R/get_sites.R

```diff
@@ -1,54 +1,58 @@
-#'
-#'@param with_child_key limit sites to those with children matching the specified key
-#'@param ... Additional parameters supplied to \code{\link[sbtools]{session_check_reauth}}
-#'@param limit numeric. Max number of sites to return. Default 10000
-#'@return a character vector of "site_root" titles (keys)
-#'@import sbtools
-#'@examples
-#'\donttest{
-#'get_sites()
-#'}
-#'\dontrun{
-#'# get sites if they are private
-#'
-#'get_sites(limit = 10)
-#'# get sites with water temperature
-#'get_sites(with_child_key = 'ts_wtr')
-#'}
-#'@import jsonlite 
-#'@import httr
-#'@import sbtools
-#'@export
-get_sites <- function(with_child_key = NULL, limit = 10000, ...){
-  
-  scheme <- get_scheme()
-  type <- 'site_root'
+#' A helper to find_sites. Gets site IDS from sciencebase w/ simple data query
+#' 
+#' Collects from sciencebase all site names ("site_root" titles, e.g., 
+#' "nwis_02948375") or the names of sites containing a particular var_src 
+#' dataset, and returns as a character vector of those site names
+#' 
+#' @param with_var_src limit sites to those with children matching the specified
+#'   var_src key
+#' @param limit numeric. Max number of sites to return. Default 10000
+#' @param ... Additional parameters supplied to 
+#'   \code{\link[sbtools]{session_check_reauth}}
+#' @return a character vector of "site_root" titles (keys)
+#' @import sbtools
+#' @examples
+#' \donttest{
+#' get_sites()
+#' }
+#' \dontrun{
+#' get_sites(limit = 10)
+#' # get those sites that have water temperature
+#' get_sites(with_var_src = 'ts_wtr')
+#' }
+#' @import jsonlite
+#' @import httr
+#' @import sbtools
+#' @export
+get_sites <- function(with_var_src = NULL, limit = 10000, ...){
   
+  # log in to ScienceBase
   session_check_reauth(...)
-  session = current_session()
 
-  #get the superset of sites, which is used for both options
-  identifiers <- query_item_identifier(scheme = scheme, type = type, session = session, limit = limit)
-  
-  if (is.null(with_child_key)){
-    sites <- identifiers$title
+  if (is.null(with_var_src)){
+    # get the superset of sites. this query is used in both if{} blocks but with
+    # different limits.
+    sites <- query_item_identifier(scheme=get_scheme(), type='site_root', limit=limit)$title
   } else {
     
-    filter_items = list('scheme'=scheme, 'type'= with_child_key)
+    # find all the time series items for the specified var_src
+    if(length(with_var_src) != 1) stop("with_var_src must have length 1")
+    # create the query
+    filter_items = list('scheme'=get_scheme(), 'type'=with_var_src)
     filter = paste0('itemIdentifier=', toJSON(filter_items, auto_unbox=TRUE))
-    
+    # run the query & pull out the content (timeseries item IDs)
     query = list('filter' = filter, 'max' = limit, 'format' = 'json', 'fields' = 'parentId')
-    child_ids <- query_items(query, session) 
+    child_ids <- query_items(query) 
```

Comment:
what i'm saying is that people had the option to pass alternative credential info in already in this function call, because several lines earlier we called `session_check_reauth(...)`. doesn't that info then become saved in the current session?